numba/tests/test_parfors.py

```diff
@@ -1957,6 +1957,120 @@ def test_impl(x, sz):
+        # Forbid fusion in unanalyzable call inside prange.
+        def test_impl(u, size):
+            for i in numba.prange(1, size - 1):
+                issue7854_proc(u, i, 0, size)
+            for i in numba.prange(1, size - 1):
+                issue7854_proc(u, i, 1, size)
+            return u
+
+        size = 4
+        u = np.zeros((size, size))
+        cptypes = (numba.float64[:, ::1], types.int64)
+        self.assertEqual(countParfors(test_impl, cptypes), 2)
+        self.check(test_impl, u, size)
+
+    def test_prange_index_calc1(self):
+        # Should forbid fusion due to cross-iteration dependency as
+        # detected by loop index calcuation (i+1) as array index.
+        def test_impl(u, size):
+            for i in numba.prange(1, size - 1):
+                for j in range((i + 1) % 2 + 1, size - 1, 2):
+                    u[i, j] = u[i + 1, j] + 1
+            for i in numba.prange(1, size - 1):
+                for j in range(i % 2 + 1, size - 1, 2):
+                    u[i, j] = u[i + 1, j] + 1
+            return u
+
+        size = 4
+        u = np.zeros((size, size))
+        cptypes = (numba.float64[:, ::1], types.int64)
+        self.assertEqual(countParfors(test_impl, cptypes), 2)
+        self.check(test_impl, u, size)
+
+    def test_prange_reverse_order1(self):
+        # Testing if reversed loop index usage as array index
+        # prevents fusion.
+        def test_impl(a, b, size):
+            for i in numba.prange(size):
+                for j in range(size):
+                    a[i, j] = b[i, j] + 1
+            for i in numba.prange(size):
+                for j in range(size):
+                    b[j, i] = 3
+            return a[0, 0] + b[0, 0]
+
+        size = 10
+        a = np.zeros((size, size))
+        b = np.zeros((size, size))
+        cptypes = (numba.float64[:, ::1], numba.float64[:, ::1], types.int64)
+        self.assertEqual(countParfors(test_impl, cptypes), 2)
+        self.check(test_impl, a, b, size)
+
+    def test_prange_parfor_index_then_not(self):
+        # Testing if accessing an array first with a parfor index then
+        # without will prevent fusion.
+        def test_impl(a, size):
+            b = 0
+            for i in numba.prange(size):
+                a[i] = i
+            for i in numba.prange(size):
+                b += a[5]
+            return b
+
+        size = 10
+        a = np.zeros(size)
+        cptypes = (numba.float64[:], types.int64)
+        self.assertEqual(countParfors(test_impl, cptypes), 2)
+        self.check(test_impl, a, size)
+
+    def test_prange_parfor_index_tuple1(self):
+        # Testing if accessing a tuple with prange index
+        # and later with a constant will not prevent fusion.
+        def test_impl(a, c, size):
+            b = 0
+            for i in numba.prange(size):
+                a[i] = i + c[i]
+            for i in numba.prange(size):
+                b += a[i] + c[1]
+            return b
```

Comment:
```suggestion
        def test_impl(a, tup, size):
            acc = 0
            for i in numba.prange(size):
                a[i] = i + tup[i]
            for i in numba.prange(size):
                acc += a[i] + tup[1]
            return acc
```
maybe this makes this test a little easier to read, the arguments this test is called with are `a, b, size` but the signature uses names `a, c, size` and then `b` is an accumulator.